### DISCUSSION TITLE
selecting from view results in error "permission denied for schema vault" while

### DISCUSSION THREAD

**Discussion by srichint1** [supabase/supabase#20697] [Questions]
Hi
I have a stripe schema where I have few Foreign tables mapped to Stripe tables (from Stripe system)
I have a view "invoiceinfo" (in custom schema called reporting) which refers to one of the above Foreign tables.
I can use the SELECT statement from the above "invoiceinfo" using role "postgres". But when I try to run a select statement using "anon" role I am getting error  "permission denied for schema vault". I tried giving  below permissions, but no use, I am getting same error
GRANT USAGE ON SCHEMA stripe TO anon, authenticated, service_role;
GRANT ALL ON ALL TABLES IN SCHEMA stripe TO anon, authenticated, service_role;

Please suggest how this can be resolved

Thanks
![vaulterr](https://github.com/supabase/supabase/assets/120005578/fe7b71eb-4a42-429b-9e1f-874c75319dd7)

**Comment by snikch** [comment]
I have come across this issue too. Please nobody do what the op has suggested here. This appears to be allowing anyone to have full access to the attached stripe account.
